src/strings.ts

```diff
@@ -127,3 +127,19 @@ const v6 = `
 export const CIDR = PatternString(`(?:^${v4}$)|(?:^${v6}$)`);
 
 export type CIDR = t.TypeOf<typeof CIDR>;
+
+/**
+ * A valid Fiscal Number (for persons)
+ */
+export const FiscalCode = PatternString(
+  "^[A-Z]{6}[0-9LMNPQRSTUV]{2}[ABCDEHLMPRST][0-9LMNPQRSTUV]{2}[A-Z][0-9LMNPQRSTUV]{3}[A-Z]$"
+);
+
+export type FiscalCode = t.TypeOf<typeof FiscalCode>;
+
+/**
+ * A valid Fiscal Number (for organizations)
+ */
+export const OrganizationFiscalCode = PatternString("^[0-9]{11}$");
```

Comment:
english name for PIVA is `VatNumber` or `VatId`

Reply:
then it must be changed here as well:

https://github.com/teamdigitale/digital-citizenship-functions/blob/327442dc144334b5d91adae65be231d4c9a4100f/lib/models/service.ts#L40

Reply:
But that's a different definition, that's a `FiscalCode` (it follows the fiscal code pattern) - while this one it's all numbers, it's indeed a PIVA (that has a different pattern than CF).

I think the confusion here is caused by the subject, they're both fiscal code (aka tax ID) but:

* the FiscalCode like FRLFRC... is a personal tax identification number
* the PIVA 12347... is a company tax identification number

what we should provide here is the ID format (Codice Fiscale or Partita Iva), then it's up to the user of the type to associate it to the right entity

I think it makes sense to use an already defined terminology (e.g. Partita IVA) than create a new one (e.g. Codice Fiscale dell'Organizzazione)

Reply:
this one (PIVA) must match the organizationFiscalCode (which is the "Codice Fiscale dell'Organizzazione" that we save into the service metadata). That's why I'm suggesting to rename organizationFiscalCode to PIVA also in the notification APIs *and* to [fix the format](https://www.pivotaltracker.com/story/show/158476749).